lib/govuk_publishing_components/presenters/public_layout_helper.rb

```diff
@@ -69,20 +100,43 @@ def footer_track_actions
         FOOTER_TRACK_ACTIONS
       end
 
-      def generate_data_attribute(link, track_action)
+      def generate_data_attribute(link, track_action, ga4_attributes)
+        if ga4_attributes
```

Comment:
Is this check needed? I can't see where this method is called without a `ga4_attributes` parameter.

Reply:
@andysellick Good point, I think the UA tests were failing when I started writing the tests, so I put this in, but then realised the UA tests need the GA4 stuff anyway so this can probably be removed.